sdk/storage/inc/common/account_sas_builder.hpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+#include "common/constants.hpp"
+#include "common/storage_credential.hpp"
+
+namespace Azure { namespace Storage {
+
+  enum class SasProtocol
+  {
+    HttpsAndHtttp,
+    HttpsOnly,
+  };
+
+  enum class AccountSasResource
+  {
+    None = 0,
+    Service = 1,
+    Container = 2,
+    Object = 4,
+    All = ~0,
+  };
+
+  inline AccountSasResource operator|(AccountSasResource lhs, AccountSasResource rhs)
```

Comment:
I personally don't like operator overloading outside of the class. But it's minor.

Reply:
enum class is not class, they cannot have member functions. Where else do you want to put them?

Reply:
Look forward to the Core's extensible enum, if that solves this problem.